pan3d/dataset_builder.py

```diff
@@ -172,6 +175,11 @@ def set_dataset_path(self, dataset_path, **kwargs):
         if len(self.state.data_attrs) > 0:
             self.state.show_data_attrs = True
         self.state.coordinates = []
+        self.state.x_array = None
+        self.state.y_array = None
+        self.state.z_array = None
+        self.state.t_array = None
+        self.state.t_index = 0
```

Comment:
Can those state var be initialized by the UI? Or the pieces that need/expect them?

Reply:
They could be initialized by the UI, but we also want to clear their values every time the selected dataset is changed. Having these statements here is both an initialization and a reset. We could copy these statements to initialize them in the UI, but it might be redundant since we need these here too.

Reply:
ok sound good.

Maybe you can condense it like

`self.state.update(dict(y_array=None, z_array=None, t_array=None, t_index=0))`

Reply:
Ok, changed in [1cb2197](https://github.com/Kitware/pan3d/pull/30/commits/1cb21978c4018dc1f92cc2e468f5ab332190ce7a)